backend/src/v2/driver/driver.go

```diff
@@ -665,6 +665,33 @@ func extendPodSpecPatch(
 		podSpec.Volumes = append(podSpec.Volumes, ephemeralVolume)
 		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, ephemeralVolumeMount)
 	}
+
+	// EmptyDirMounts
+	for _, emptyDirVolumeSpec := range kubernetesExecutorConfig.GetEmptyDirMounts() {
+		var sizeLimitResource *k8sres.Quantity
+		if emptyDirVolumeSpec.GetSizeLimit() != "" {
+			r := k8sres.MustParse(emptyDirVolumeSpec.GetSizeLimit())
+			sizeLimitResource = &r
+		}
+
+		emptyDirVolume := k8score.Volume{
+			Name: emptyDirVolumeSpec.GetVolumeName(),
+			VolumeSource: k8score.VolumeSource{
+				EmptyDir: &k8score.EmptyDirVolumeSource{
+					Medium:    k8score.StorageMedium(emptyDirVolumeSpec.GetMedium()),
+					SizeLimit: sizeLimitResource,
+				},
+			},
+		}
+		emptyDirVolumeMount := k8score.VolumeMount{
+			Name:      emptyDirVolumeSpec.GetVolumeName(),
+			MountPath: emptyDirVolumeSpec.GetMountPath(),
+		}
+
+		podSpec.Volumes = append(podSpec.Volumes, emptyDirVolume)
+		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, emptyDirVolumeMount)
```

Comment:
The code assumes that `Containers[0]` is always valid and the target container for mounting the volume. If `podSpec.Containers` is empty or if there are multiple containers in the pod, this could lead to errors or incorrect behavior. 
Consider adding a check to ensure the container exists or iterating through all containers if the volume should be mounted to more than one. Something like:

```
if len(podSpec.Containers) == 0 {
    return fmt.Errorf("podSpec has no containers to mount volumes to")
}
```

Reply:
had to refresh my memory on this :)

I copied this block from the other blocks already in driver. I think you're exactly right and hardcoding the `[0]` is not great, but I want to tackle all of them in a refactor PR that we can do later.

Examples: [1](https://github.com/kubeflow/pipelines/blob/2e6e634de43c42fee88dfe2abfdb0e4155dc4f95/backend/src/v2/driver/driver.go#L487-L488), [2](https://github.com/kubeflow/pipelines/blob/2e6e634de43c42fee88dfe2abfdb0e4155dc4f95/backend/src/v2/driver/driver.go#L505-L506), [3](https://github.com/kubeflow/pipelines/blob/2e6e634de43c42fee88dfe2abfdb0e4155dc4f95/backend/src/v2/driver/driver.go#L565) ... you get the idea.